core/src/main/resources/swagger/api-docs.yaml

```diff
@@ -5625,6 +5625,9 @@ components:
         workspaceVersion:
           type: string
           description: internal use
+        billingAccount:
+          type: string
+          description: billing account being used for the workspace
```

Comment:
I'll probably make this comment at other places too, but I wonder if we can add to this description to clarify that this field on the `Workspace` class is not really the "source of record" for the desired Billing Account for the Workspace.  This field is sort of a locally cached value for the Billing Account that Workspace is currently billing to, but the _desired/intended_ Billing Account is actually specified on the Billing Project.  The value in this field should match the Billing Project's Billing Account value, and if they do not match, the Billing Account Synchronizer cron job should modify the Workspace's Google Project and this field to match the Billing Project's Billing Account.  

I would like to see these comments and descriptive text in Swagger and at least on the `WorkspaceModel`.  I want to make it really clear/obvious to anyone who looks at the `Workspace` class (companion object, and all the other slight variations of Workspace) that in almost all scenarios they should really just ignore this field and look at the `Workspace.billingProject.billingAccount` instead.  

AND/OR - maybe rename this field, `currentBillingAccountOnGoogleProject` or something like that.  I think that's more precise about what this field is and what it's used for and might make all of this disambiguation between `Workspace.billingAccount` and `BillingProject.billingAccount` a lot easier.  Also makes it way harder for a user to use the wrong field if they have different names.

Reply:
I prefer renaming a field to adding comments, since comments can easily go out of date, so I will do the renaming if that works!